### PR TITLE
feat:[#3115] show suspended objects

### PR DESCRIPTION
**Pull request by goreck888**
Add entities's application services
Remove code duplication
Add `suspended` and `draft` statuses
Unify `statusable` logic

Closes https://github.com/cyfronet-fid/marketplace/issues/3115

### CODE REVIEW

**Comment by Marcelinna** [comment]
I can click on link in right box
image

**Comment by Marcelinna** [comment]
in markteplace I dont see offer and bundle for suspended service

**Comment by Marcelinna** [comment]
When I click on Provider tab in BO
image

**Comment by agpul** [comment]
https://www.figma.com/file/VEJHiowgFv2sKfCTaPrUGf/EOSC---Ala?type=design&mode=design&t=tN4t6C19AQG4nybU-0

Adjust changes according to the Figma. Have a deep look which fields should be active and which should not.

**Comment by Marcelinna** [comment]
- Pricing is stil active
- Monitoring is active
image

**Comment by Marcelinna** [comment]
About
image
Details - everything is still active
image
image

**Comment by Marcelinna** [comment]
interoperability guidelines button is still active
image

**Comment by Marcelinna** [comment]
still active
obraz

**Comment by Marcelinna** [comment]
Published 
obraz

Suspended - disappear Uptime monitoring, Statistics, Availability and Language
obraz

**Comment by Marcelinna** [comment]
it is not posible find suspended service in backoffice, should be it should be found like draft
obraz

**Comment by Marcelinna** [comment]
still active
obraz

**Comment by AlicjaSwierad** [comment]
- Active links should be in color: #010E87
- The provider section should be active - 
image

**Comment by agpul** [comment]
https://www.figma.com/file/VEJHiowgFv2sKfCTaPrUGf/EOSC---Ala?type=design&node-id=2128-37659&mode=design

**Comment by eldenkat** [comment]
Bundle cannot be unpublished. Please ensure your form is properly completed.

**Review by Marcelinna — APPROVED** [review]

File: app/policies/backoffice/service_policy.rb [hunk 1]
```diff
@@ -46,15 +46,23 @@ def update?
...
+    service_portfolio_manager? && !record.published? && !record.deleted?
   end
 
   def publish_unverified?
-    service_portfolio_manager? && (record.draft? || record.published?) && !record.deleted?
+    service_portfolio_manager? && !record.unverified? && !record.deleted?
+  end
+
+  def suspend?
+    service_portfolio_manager? && !record.suspended? && !record.deleted?
```
[the remark this answers is not in the document]
Comment (goreck888) [reply app/policies/backoffice/service_policy.rb:57]: Unverified records can be set to published or suspended, and errored has some validation errors, so it calls an error when trying to publish

File: app/policies/bundle_policy.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ class Scope < Scope
     def resolve
       scope
         .joins(:service)
-        .where("bundles.status = ? AND services.status IN (?)", "published", %w[published unverified])
+        .where("bundles.status = ? AND services.status IN (?)", "published", %w[published unverified suspended errored])
```
[the remark this answers is not in the document]
Comment (goreck888) [reply app/policies/bundle_policy.rb]: Done.

**Pull request merged**